OnBoardingCalculator/app/src/main/java/com/globant/onboardingcalculator/mvp/presenter/CalculatorPresenter.java

```diff
@@ -2,36 +2,111 @@
+import static com.globant.onboardingcalculator.utils.Constants.NUMBER_ZERO;
+import static com.globant.onboardingcalculator.utils.Constants.OPERATOR_DIVIDE;
+import static com.globant.onboardingcalculator.utils.Constants.OPERATOR_MULTIPLY;
+import static com.globant.onboardingcalculator.utils.Constants.OPERATOR_PLUS;
+import static com.globant.onboardingcalculator.utils.Constants.OPERATOR_SUBSTRACTION;
+import static java.lang.Double.parseDouble;
+
 
 public class CalculatorPresenter {
 
     private CalculatorModel model;
     private CalculatorView view;
+    private String result;
+    private DecimalFormat decimalFormat = new DecimalFormat("#.##");
 
     public CalculatorPresenter(CalculatorModel model, CalculatorView view) {
         this.model = model;
         this.view = view;
+        result = EMPTY_STRING;
     }
 
     public void onClearPressed() {
+        result = EMPTY_STRING;
         model.clearOperation();
         view.clearVisor();
     }
 
     public void onOperatorPressed(char operator) {
-        if (model.getOperator() == Constants.EMPTY_CHAR)
+        if (!model.getFirstOperand().isEmpty()) {
             model.setOperator(operator);
-        view.refreshVisor(String.valueOf(operator));
+            view.refreshVisor(String.valueOf(operator));
+            view.enablePointBtn();
+        } else
+            view.showOperatorError();
     }
 
     public void onNumberPressed(String number) {
-        if (model.getOperator() == Constants.EMPTY_CHAR) {
+        if (!result.equals(EMPTY_STRING) && (model.getOperator() == EMPTY_CHAR))
+            view.showOperatorErrorAfterEqualPressed();
+        else if (model.getOperator() == EMPTY_CHAR) {
             model.setFirstOperand(number);
             view.refreshVisor(model.getFirstOperand());
         } else {
             model.setSecondOperand(number);
             view.refreshVisor(model.getSecondOperand());
         }
     }
-}
\ No newline at end of file
+
+    public void onEqualPressed() {
+        if (!model.emptyOperation())
+            switch (model.getOperator()) {
+                case OPERATOR_PLUS:
+                    result = String.valueOf(decimalFormat.format(parseDouble(model.getFirstOperand()) + parseDouble(model.getSecondOperand())));
+                    model.operate(result);
+                    break;
+                case OPERATOR_SUBSTRACTION:
+                    result = String.valueOf(decimalFormat.format(parseDouble(model.getFirstOperand()) - parseDouble(model.getSecondOperand())));
+                    model.operate(result);
+                    break;
+                case OPERATOR_MULTIPLY:
+                    result = String.valueOf(decimalFormat.format(parseDouble(model.getFirstOperand()) * parseDouble(model.getSecondOperand())));
+                    model.operate(result);
+                    break;
+                case OPERATOR_DIVIDE:
+                    if (parseDouble(model.getSecondOperand()) != parseDouble(NUMBER_ZERO)) {
```

Comment:
Nico i did in this way because i wanted to contemplate the case of zero division, so i thought that it was a specific check that must only be in the division case. Thanks Nico!